File: practica1-27-08/estadistica.py
import math
print("Ingrese 10 números separados por espacios:")
n = input().split()

numeros = []
for num_str in n:
    numeros.append(float(num_str))

s = 0
for num in numeros:
    s += num
p = s / len(numeros)

scuadrados= 0
for i, num in enumerate(numeros):
    d = num - p
    cuad = d*d
    scuadrados += cuad

desviacion = math.sqrt(scuadrados / (len(numeros) - 1))

print(f"\nEl promedio es {promedio:.2f}")
print(f"La desviación estándar es {desviacion:.5f}")

File: practica1-27-08/Ejercicioo3/src/Estadistica.java

public class Estadistica {
    private double[] numeros;
    public Estadistica(double[] numeros) {
        this.numeros = numeros;
    }

    public double promedio() {
        double suma = 0;
        for (int i = 0; i < numeros.length; i++) {
            suma += numeros[i];
        }
        return suma / numeros.length;
    }

    public double desviacion() {
        double prom = promedio();
        double sumaCuadrados = 0;

        for (int i = 0; i < numeros.length; i++) {
            double diferencia = numeros[i] - prom;
            sumaCuadrados += diferencia * diferencia;
        }
        return Math.sqrt(sumaCuadrados / (numeros.length - 1));
    }
}
